org.eclipse.scanning.server/src/org/eclipse/scanning/server/servlet/ScanProcess.java

```diff
@@ -209,6 +175,61 @@ public void execute() throws EventException {
 		}
 	}
 
+	private void executeNonBlocking(IDeviceController controller, ScanBean bean) throws ScanningException, InterruptedException, TimeoutException, ExecutionException {
+		
+		logger.debug("Running non-blocking device {}", controller.getDevice().getName());
+		controller.getDevice().start(null);
+		
+		long latchTime = Long.getLong("org.eclipse.scanning.server.servlet.asynchWaitTime", 500);
+		logger.debug("Latching on device {} for {}", controller.getDevice().getName(), latchTime);
+		controller.getDevice().latch(latchTime, TimeUnit.MILLISECONDS); // Wait for it to do a bit in case of errors.
+		
+		logger.warn("Cannot run end script when scan is async. (Scan has not been cancelled, after script has been ignored.)");
+		logger.warn("Cannot perform end position when scan is async. (Scan has not been cancelled, end has been ignored.)");
+	}
+
+	private void executeBlocking(IDeviceController controller, ScanBean bean) throws ScanningException, InterruptedException, TimeoutException, ExecutionException, EventException, UnsupportedLanguageException, ScriptExecutionException {
+		
+		logger.debug("Running blocking controller {}", controller.getName());
+		controller.getDevice().run(null); // Runs until done
+	    
+		// Run a script, if any has been requested
+		runScript(bean.getScanRequest().getAfter(), bean.getScanRequest()::setAfterResponse);
+        setPosition(bean.getScanRequest().getEnd(), "end");
+	}
+
+	private void setPosition(IPosition pos, String location) throws ScanningException, InterruptedException {
+		if (pos!=null) {
+			IPosition start = bean.getScanRequest().getStart();
+			positioner.setPosition(start);
+			logger.debug("The "+location+" position {} is reached.", start);
+		}
+	}
+
+	private void validateRequest(ScanBean bean) throws ValidationException, InstantiationException, IllegalAccessException {
+		if (!Boolean.getBoolean("org.eclipse.scanning.server.servlet.scanProcess.disableValidate")) {
+			logger.debug("Validating run : {}", bean);
+			final ScanRequest<?> sr = bean.getScanRequest();
+			if (sr.getDetectors()!=null && sr.getDetectors().isEmpty()) sr.setDetectors(null);
+		    Services.getValidatorService().validate(sr);
+			logger.debug("Validating passed : {}", bean);
+		} else {
+			logger.warn("The run {} has validation switched off.", bean);
+		}
+	}
+
+	private void checkMonitors() throws Exception {
+		
+		// We set any activated monitors in the request if none have been specified.
+		if (Boolean.getBoolean("org.eclipse.scanning.server.useDefaultActivatedMonitors")) {
+			if (bean.getScanRequest().getMonitorNames()==null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1066)